src/main/scala/com/gu/support/workers/lambdas/GetRecurringSubscription.scala

```diff
@@ -0,0 +1,46 @@
+package com.gu.support.workers.lambdas
+
+import java.time.LocalDateTime
+
+import cats.implicits._
+import com.gu.support.workers.model.BillingPeriod
+import com.gu.zuora.GetAccountForIdentity.DomainAccount
+import com.gu.zuora.ZuoraConfig.RatePlanId
+import com.gu.zuora.ZuoraService
+import com.gu.zuora.model.response.{RatePlan, Subscription}
+
+import scala.concurrent.ExecutionContext.Implicits.global
+import scala.concurrent.Future
+
+object GetRecurringSubscription {
+
+  def apply(zuoraService: ZuoraService, now: LocalDateTime, identityId: String, billingPeriod: BillingPeriod): Future[Option[Subscription]] = {
```

Comment:
I've just split this from the zuora service as it was just packing that class out with unrelated code and making it harder to test.  It shouldn't really depend on everything inside zuoraService object, but that kind of decoupling is not a task to be taken lightly in this PR.